### PR TITLE
Add new "Accordion" component with customizable Inputs

### PR DESCRIPTION
**Pull request by deeved-hiuston-brisa**
## Issue Number

#849 

## Description

This Pull Request aims to add a new component called "Accordion" to our project. The "Accordion" component provides the ability to hide and show content with a collapse/expand feature. It comes with a set of customizable Inputs for greater usage flexibility.

## Proposed Changes

1. Addition of the "Accordion" component to the components directory.
2. Definition of the following Inputs for the "Accordion" component:
- @Input() name?: string. This Input allows associating a name with the component, making it more identifiable and useful in contexts where multiple "Accordion" components may be used.
- @Input() templateHeader: TemplateHeader. This Input allows providing a custom TemplateHeader for the header of the "Accordion" component, giving the freedom to customize the header's appearance.
- @Input() show? = false. This Input controls the display state of the content within the "Accordion" component. When set to true, the content is displayed; otherwise, it remains hidden by default.
3. Creation of unit tests to ensure the correct functioning of the "Accordion" component and its customizable Inputs.
4. Addition of comprehensive documentation using Storybook for the "CoAccordionlapse" component, demonstrating how to use it with examples and usage scenarios.

## Motivation

The addition of this new "Accordion" component with customizable Inputs will enhance the flexibility and usability of our project, enabling developers to create more dynamic and appealing user interfaces.

## How to Test

To test the "Accordion" component, please follow the instructions in the Storybook and utilize the provided unit tests in this Pull Request.

## View Storybook

[Accordion](https://62eab350a45bdb0a5818520e-ttqvyhdhza.chromatic.com/?path=/story/ion-data-display-accordion--accordion)

### CODE REVIEW

**Review by alysson-mascarenhas-brisa — APPROVED** [review]
It was very good and easy to use. Congratulations on the work!

File: projects/ion/src/lib/accordion/accordion.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    expect(screen.getByTestId('ion-accordion')).toBeTruthy();
+  });
+
+  it('should render ion-accordion with name Brisanet', async () => {
+    accordionTestComponent.name = 'Brisanet';
+    fixture.detectChanges();
+    expect(screen.getByTestId('ion-accordion__header-name')).toHaveTextContent(
+      'Brisanet'
+    );
+  });
```
Comment (iurynogueira) [inline projects/ion/src/lib/accordion/accordion.component.spec.ts:65]:
```suggestion
  it('should render ion-accordion with name Brisanet', async () => {
    const accordionName = 'Brisanet';
    accordionTestComponent.name = accordionName ;
    fixture.detectChanges();
    expect(screen.getByTestId('ion-accordion__header-name')).toHaveTextContent(
      accordionName 
    );
  });
```

File: projects/ion/src/lib/accordion/accordion.component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { TemplateHeader } from '../core/types/accordion';
+
+@Component({
+  selector: 'ion-accordion',
+  templateUrl: './accordion.component.html',
+  styleUrls: ['./accordion.component.scss'],
+})
+export class IonAccordionComponent {
+  @Input() name?: string;
+  @Input() templateHeader: TemplateHeader;
```
Comment (iurynogueira) [inline projects/ion/src/lib/accordion/accordion.component.ts:11]:
```suggestion
  @Input() templateHeader?: TemplateHeader;
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Nice!

File: projects/ion/src/lib/accordion/accordion.component.html
[the remark this answers is not in the document]
Comment (deeved-hiuston-brisa) [reply projects/ion/src/lib/accordion/accordion.component.html:1]: Thanks for the tip! :heart:

File: projects/ion/src/lib/accordion/accordion.component.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      >
+        {{ name }}
+      </div>
+      <ng-template
+        #headerCustom
+        [ngTemplateOutlet]="templateHeader"
+      ></ng-template>
+    </div>
+    <ion-icon *ngIf="!show" type="semi-up" [size]="iconSize"></ion-icon>
+    <ion-icon *ngIf="show" type="semi-down" [size]="iconSize"></ion-icon>
```
[the remark this answers is not in the document]
Comment (deeved-hiuston-brisa) [reply projects/ion/src/lib/accordion/accordion.component.html:21]: True. Thanks for the observation. :smile:

**Review by iurynogueira — APPROVED** [review]

File: projects/ion/src/lib/accordion/accordion.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (allan-chagas-brisa) [reply projects/ion/src/lib/accordion/accordion.component.html]: No worries :D

**Pull request merged**